dodgeBall/main.js

```diff
@@ -48,35 +50,191 @@ const arrOfPeople = [
-    button.addEventListener('click', function() {makePlayer(person.id)} )
+    button.addEventListener('click', function() {
+      makePlayer(person.id)
+      listElement.removeChild(li)
+    })
     li.appendChild(button)
     li.appendChild(document.createTextNode(person.name + " - " + person.skillSet))
     listElement.append(li)
   })
 }
 
+function listPlayers(){
+  let listElement = document.getElementById('players');
+  listElement.innerHTML="";
+
+  listOfPlayers.map(person => {
+    let player = person.person[0];
+    let li = document.createElement('li');
+    let blueButton = document.createElement('button');
+    blueButton.innerHTML = 'Become a Smurf!';
+    blueButton.addEventListener('click', function(){joinBlue(player); updateDOM();});
+    li.appendChild(blueButton);
+    let redButton = document.createElement('button');
+    redButton.innerHTML = 'Become a Devil!';
+    redButton.addEventListener('click', function(){joinRed(player); updateDOM();});
+    li.appendChild(redButton);
+    li.appendChild(document.createTextNode(player.name + ' / ' + player.skillSet));
+    listElement.append(li)
+  })
+}
+const listTeams = () => {
+let listElement = document.getElementById('blue')
+listElement.innerHTML="";
+
+blueTeam.map(person => {
+  const li = document.createElement('li');
+  li.appendChild(document.createTextNode(person.name + " / " + person.skillSet));
+  listElement.appendChild(li);
+})
+listElement = document.getElementById('red');
+listElement.innerHTML="";
+
+redTeam.map(person => {
+  const li = document.createElement('li');
+  li.appendChild(document.createTextNode(person.name + ' / ' + person.skillSet));
+  listElement.append(li);
+})
+}
+function updateDOM(){
+  listPlayers();
+  listTeams();
+}
+
+function removeFromPlayersList(player){
+  let id = player.id;
+  console.log(id);
+  for (let i = 0; i < listOfPlayers.length; i++){
+    if (listOfPlayers[i].person[0].id === id){
+      listOfPlayers.splice(i, 1);
+    }
+  }
+}
+
+function joinRed(player){
+  redTeam.push(player);
+  newTeammate.joinRedTeam(player);
+  removeFromPlayersList(player);
+}
+
+function joinBlue(player){
+  blueTeam.push(player);
+  newTeammate.joinBlueTeam(player);
+  removeFromPlayersList(player);
+}
+const updateNewPlayer = (currentPerson) => {
+  let newPlayer = new Player(currentPerson, true, true, true, true, 8);
+  listOfPlayers.push(newPlayer);
+  newPlayer.addPlayer(currentPerson);
+}
+
 const makePlayer = (id) => {
-  console.log(`li ${id} was clicked!`)
-}
\ No newline at end of file
+  let currentPerson;
+  for (let i = 0; i < arrOfPeople.length; i++){
+    if (arrOfPeople[i].id === id){
+      currentPerson = arrOfPeople[i];
+      arrOfPeople.splice(i, 1);
+    }
+  }
+  updateNewPlayer(currentPerson);
+  listPlayers();
+}
+
+///////TESTS\\\\\\\
+
+//I started to do the tests, and it seemed to mess up my code... not sure why
+//I was took the code from the ship assignment and was using that as a template..
+
```

Comment:
I know these tests don't match up, I was using the mars one as a template and then couldn't get the first test to work because of the people part...